recipe/meta.yaml

```diff
@@ -1,75 +1,139 @@
-    - dir %SP_DIR%\mypy\*.pyd /s /b  # [win]
-    - ls $SP_DIR/mypy/*.so  # [linux and not (aarch64 and py==310)]
+outputs:
+  - name: mypy
+    build:
+      script:
+        - export MYPY_USE_MYPYC=1  # [not win]
+        - export SETUPTOOLS_USE_DISTUTILS=stdlib  # [not win]
+        - set MYPY_USE_MYPYC=1     # [win]
+        - set SETUPTOOLS_USE_DISTUTILS=stdlib  # [win]
+        # mypy searches for <>-stubs in site.getsitepackages()
+        # which points to $SP_DIR, even in cross-python
+        - cp -rv $BUILD_PREFIX/lib/python$PY_VER/site-packages/*-stubs $SP_DIR  # [build_platform != target_platform]
+        - cp -rv $BUILD_PREFIX/lib/python$PY_VER/site-packages/tomli* $SP_DIR  # [build_platform != target_platform]
+        - {{ PYTHON }} -m pip install . --no-deps -vv
+      entry_points:
+        - mypy = mypy.__main__:console_entry
+        - stubgen = mypy.stubgen:main
+        - dmypy = mypy.dmypy.client:console_entry
+        - stubtest = mypy.stubtest:main
+        # CLI provided here, but doesn't depend on a compiler
+        - mypyc = mypyc.__main__:main
+
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - {{ compiler('c') }}
+        # mypy's build imports this module, which has a C extension,
+        # so it has to be native to the build platform
+        - typed-ast >=1.4.0,<2                   # [build_platform != target_platform and py<38]
+        - tomli >=1.1.0
+      host:
+        - python
+        - pip
+        - mypy_extensions >=0.4.3
+        - typed-ast >=1.4.0,<2  # [py<38]
+        - typing_extensions >=3.10
+        - types-setuptools
+        - types-psutil
+        - types-typed-ast >=1.5.8,<1.6.0
+        - tomli >=1.1.0
+      run:
+        - python
+        - mypy_extensions >=0.4.3
+        - typed-ast >=1.4.0,<2  # [py<38]
+        - typing_extensions >=3.10
+        - tomli >=1.1.0
+        # extra: dmypy
+        - psutil >=4.0
+
+    test:
+      requires:
+        - pip
+      imports:
+        - mypy
+        - mypy.dmypy
+        - mypy.plugins
+        - mypy.server
+        - mypyc
+      commands:
+        - pip check
+        - dmypy --help
+        - mypy --help
+        - mypyc --help
+        - stubgen --help
+        - stubtest --help
+        - dir %SP_DIR%\mypy\*.pyd /s /b  # [win]
+        - ls $SP_DIR/mypy/*.so  # [linux and not (aarch64 and py==310)]
+
+  - name: mypyc
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage("mypy", exact=True) }}
+        - gcc_linux-64     # [linux]
+        - clang_osx-64     # [osx]
+        - m2w64-toolchain  # [win]
+    test:
+      files:
+        - test_mypyc.py
+      requires:
+        - pip
+      commands:
+        - pip check
+        - mypyc --help
+        - python test_mypyc.py  # [not(aarch64 or ppc64le)]
```

Comment:
not proud of this, but didn't know what else to do

Reply:
It is more common than we would like to admit. No judgments here ;-p

Reply:
here's [a fail](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=609636&view=logs&j=bb1c2637-64c6-57bd-9ea6-93823b2df951&t=350df31b-3291-5209-0bb7-031395f0baa1&l=7528) if it jostles anything loose in a reviewer's mind

```
running build_ext
building 'fib' extension
creating build/temp.linux-aarch64-cpython-310
creating build/temp.linux-aarch64-cpython-310/build
$PREFIX/bin/x86_64-conda-linux-gnu-cc -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O3 -Wall -fPIC -O3 -isystem $PREFIX/include -fPIC -O3 -isystem $PREFIX/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/mypyc-0.990 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $PREFIX/include -fPIC -I$PREFIX/lib/python3.10/site-packages/mypyc/lib-rt -I$PREFIX/include/python3.10 -c build/__native.c -o build/temp.linux-aarch64-cpython-310/build/__native.o -O3 -g1 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
creating build/lib.linux-aarch64-cpython-310
$PREFIX/bin/x86_64-conda-linux-gnu-cc -shared -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/mypyc-0.990 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $PREFIX/include build/temp.linux-aarch64-cpython-310/build/__native.o -o build/lib.linux-aarch64-cpython-310/fib.cpython-310-aarch64-linux-gnu.so
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/../lib/gcc/x86_64-conda-linux-gnu/12.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: skipping incompatible /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
copying build/lib.linux-aarch64-cpython-310/fib.cpython-310-aarch64-linux-gnu.so -> 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: /home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/test_tmp/fib.cpython-310-aarch64-linux-gnu.so: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/test_tmp/test_mypyc.py", line 23, in <module>
    subprocess.check_call([sys.executable, "-c", "import fib"])
  File "/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/conda/feedstock_root/build_artifacts/mypy-split_1668016285943/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python', '-c', 'import fib']' returned non-zero exit status 1.
```

Reply:
I think we have this sorted with `c-compiler`!